src/App.tsx

```diff
@@ -1,49 +1,123 @@
 import './App.css';
-import './SectionView.css';
+import './css_reset.css'
+import './section_view.css';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faFaceSmile } from '@fortawesome/free-solid-svg-icons';
 
 function App() {
-  return (
-    <div className="parent">
-      <div className="section-title">
-        <div className='text'>
-          <h1> Intro </h1>
-          <p>This intro section consists of a tuba quartet in the style of DJ Templeton & The Windsurfers</p>
-        </div>
-        <div className='buttons'>
-          <button> 42 revisions </button>
-          <button> save revision </button>
-        </div>
-      </div>
-      <div className="chords">
-        <ol>
-          <li>C</li>
-          <li>Dm</li>
-          <li>F</li>
-          <li>G</li>
-        </ol>
-      </div>
-      <div className="files">
-        <span>+ Files</span>
-          <div>Bass.mp3 <br></br> <br></br> 2 Comments</div>
-          <div>Drums.mp3 <br></br> <br></br> 2 Comments</div>
-          <div>Yodeling.mp3 <br></br> <br></br> 2 Comments  </div>
-          <div>Tuba.mp3 <br></br> <br></br> 2 Comments  </div>
-      </div>
-      <div className="comments">
-        <span>3 Comments</span>
-        <div className="display-comments">
-        <p><FontAwesomeIcon icon={faFaceSmile} /> Jerry: My name is Schmoopie and I love this song. It reminds me of my grandpa</p>
-        <p><FontAwesomeIcon icon={faFaceSmile} /> Larry: jokes on you! Im deaf.</p>
-        <p><FontAwesomeIcon icon={faFaceSmile} /> Terry: Notice the tinnitus ringing in your ears, let the hiss from your damaged hearing remind you of peaceful ocean waves...</p>
-        </div>
+
+
+
+  const sectionData = {
+    name: 'Intro',
+    description: 'This intro section consists of a tuba quartet in the style of DJ Templeton & The Windsurfers',
+    numRevisions: 42,
+  };
+
+  const currentChordProgression = ['C', 'Dm', 'F', 'G']
+
+  const files = [
+    {
+      title: 'Bass.mp3',
+      numComments: 2,
+    },
+    {
+      title: 'Drums.mp3',
+      numComments: 2,
+    },
+    {
+      title: 'Yodeling.mp3',
+      numComments: 2,
+    },
+    {
+      title: 'Tuba.mp3',
+      numComments: 2,
+    },
+  ]
+
+
+  const comments = [
+    {
+      name: 'Jerry',
+      commentText: 'My name is Schmoopie and I love this song. It reminds me of my grandpa'
+    },
+    {
+      name: 'Lerry',
+      commentText: 'jokes on you! Im deaf.'
+    },
+    {
+      name: 'Jerry',
+      commentText: 'Notice the tinnitus ringing in your ears, let the hiss from your damaged hearing remind you of peaceful ocean waves...'
+    },
+  ]
+
+
+  // Components
+
+  function Files() {
```

Comment:
We have all of the components defined inside of this `App` component, as well as the creation of the test data happening inside of the component. This has a side effect of making it so React has no context of when things "change" here, or when they have "stayed the same" between renders.

The reference to the child component definitions are changing every time, and the data being passed in as props (at least they likely will be later). For minimal refactoring to improve this, I recommend the following making the following changes:

Create `types.ts`

```ts
export type FileInfo = {
  id: string;
  // ...
}

// Other types...
```

Create `testdata.ts`

```ts
import {
  FileInfo,
  // Other types...
} from './types';

const files: FileInfo[] = [
  {
    id: '1',
    title: 'my_file_name'.mp3,
    // ...
  },
  // ...
];

const comments: Comment[] = [

];

export const testData = {
  // this is shorthand for `files: files`
  files,
  comments,
}
```

Then in `App.tsx`

```tsx
import {
  FileInfo,
  Comment,
} from './types';

import {testData} from './testdata';

function App() {
  return (
    <div className="root">
      <SectionTitle sectionData={testData.sectionData} />
      <ChordProgression chordProgression={testData.chordProgression} />
      <Files files={testData.files} />
      <Comments comments={testData.comments} />
      <Submit />
    </div>
  );
}

// Files component

type FilesProps = {
  files: FileInfo
}

function Files(props: FileProps) {
  // then use `props.files` to render things
} 
```


We define components like `Files` _outside_ of the definition of `App`. Ideally these are defined in their own files, but keeping them in the same file as `App` is less work for now. Separating the definitions to be on the module-level of the file is important though.

Reply:
Okay, so do you want me to move the components to their own files?

Reply:
Also vscode and typescript make it really easy to move components into other files. You can highlight the text of the component, then right-click and select "Refactor..." (I prefer to use the command palette and type "refactor" instead of right-clicking), then select "Move to a new file". It generates a new file, which you can then rename and relocate as necessary. vscode will automatically:

- Move that component's function to a new file, and export it as a named export
- Import that component in the original file
- Move/copy and necessary imports (like React) into the new file, and remove them from this file if they are no longer used anywhere in this file

You don't have to do any of these things. It just does it for you. It's called "refactor" here since the change is made cleanly, and the functionality of the program doesn't change

Reply:
Yeah man, I love the refactor command! It's so useful